ruint-macro/src/lib.rs

```diff
@@ -91,13 +91,12 @@ fn parse(value: &str, bits: &str) -> Result<TokenStream, String> {
         #[allow(clippy::cast_lossless)]
         if digit > base as u64 {
             return Err(format!(
-                "Invalid digit {} in base {} (did you forget the `0x` prefix?)",
-                c, base
+                "Invalid digit {c} in base {base} (did you forget the `0x` prefix?)"
             ));
         }
 
-        // Multiply result by base
-        let mut carry = 0_u64;
+        // Multiply result by base and add digit
+        let mut carry = digit;
```

Comment:
Turns out we can trivially merge it with the basis multiplication :)

Reply:
That's amazing! So much cleaner :)